backend/handlers/post_handler.py

```diff
@@ -90,14 +78,10 @@ def patch(self, user, post_urlsafe):
 
         post = ndb.Key(urlsafe=post_urlsafe).get()
 
-        Utils._assert(post.has_activity(),
+        Utils._assert(not post.can_edit(user),
```

Comment:
is better separate, because one thing is "the post cannot be edited" and another thing is "you don't have permission to edit this post"